Zodo integration

amagge opened the pull request:
Ready to add integration changes to the master branch.

developerDemetri commented:
👀

developerDemetri commented on src/main/java/edu/asu/zoophy/rest/database/LocationsRSExtractor.java:
```diff
@@ -0,0 +1,36 @@
+ * @author amagge
+ */
+public class LocationsRSExtractor implements ResultSetExtractor<List<PossibleLocation>> {
+	public static final Logger logger = org.slf4j.LoggerFactory.getLogger(LocationsRSExtractor.class);
```
Might be worth updating all the existing logging to use slf4j instead of `util.logging` ¯\_(ツ)_/¯
Also, can't this be shortened to `LoggerFactory.getLogger(LocationsRSExtractor.class);` and an import?

developerDemetri replied:
Lol I think slf4j is better if you wanted to use  that instead. I was just lazy and used the `util.logging`

developerDemetri commented on src/main/java/edu/asu/zoophy/rest/database/LocationsRSExtractor.java:
```diff
@@ -0,0 +1,36 @@
+	public static final Logger logger = org.slf4j.LoggerFactory.getLogger(LocationsRSExtractor.class);
+	
+	@Override
+	public List<PossibleLocation> extractData(ResultSet rs) throws SQLException {
```
Any particular reason to use a `ResultSetExtractor` instead of a `RowMapper`?
Should work the same for `jdbc.query`, like https://github.com/ZooPhy/zoophy-services/blob/89685d0c1f64fedcc15d62acc426da7407e441b8/src/main/java/edu/asu/zoophy/rest/database/ZooPhyDAO.java#L105 
Would be nice to keep the convention.

amagge replied:
I remember having problems with RowMapper for the list when I wrote it (but funnily not for genes which is a list too). ResultSetExtractor seemed to work fine then. I'll check it again because I've forgotten why it didn't work. ;)

developerDemetri replied:
Should behave the same https://stackoverflow.com/questions/10074025/what-is-difference-between-resultsetextractor-vs-rowmapper

developerDemetri commented on src/main/java/edu/asu/zoophy/rest/database/ZooPhyDAO.java:
```diff
@@ -70,6 +72,7 @@ public GenBankRecord retrieveLightRecord(String accession) throws GenBankRecordN
 			final String[] parameters = {accession};
 			try {
 				record = jdbc.queryForObject(PULL_RECORD_DETAILS, parameters, new GenBankRecordRowMapper());
+				record.setPossibleLocations(jdbc.query(PULL_RECORD_POSSIBLE_LOCATIONS, parameters, new LocationsRSExtractor()));
```
I think this belongs in `retrieveFullRecord`.
`retrieveLightRecord` was meant to avoid multiple SQL calls for listing basic details of search results, while `retrieveFullRecord` gets all the associated details needed (Genes, etc.) for various processing https://github.com/ZooPhy/zoophy-services/blob/89685d0c1f64fedcc15d62acc426da7407e441b8/src/main/java/edu/asu/zoophy/rest/database/ZooPhyDAO.java#L99

amagge replied:
Ah, light was to avoid multiple queries. I thought it was just "light" for the data. I'll add it to the Full and change the UI call.

developerDemetri commented on src/main/java/edu/asu/zoophy/rest/pipeline/ZooPhyRunner.java:
```diff
@@ -26,7 +26,8 @@
 	public ZooPhyRunner(String replyEmail, String jobName, boolean useGLM, Map<String, List<Predictor>> predictors, XMLParameters xmlOptions) throws PipelineException {
-		final String id  = UUID.randomUUID().toString();
+		char rand_char = (char) (Math.random()*26 + 'a');				
+		final String id  = rand_char + UUID.randomUUID().toString();	//Used as property by SpreaD3, hence start with char
```
Nice catch, never looked into the random SpreaD3 errors.

amagge commented:
Added the following changes,
1) changed Logger to util.logger
2) changed RSExtractor to RowMapper.
3) moved possibel locations query to full record implementation.

amagge commented:
Thank you for the review. I really appreciate it.

developerDemetri commented:
Happy to help. Almost forgot, need to uprev the version to `0.1.2`. Went ahead and took care of it via https://github.com/ZooPhy/zoophy-services/commit/ae14aed19d48701bf8aafc6f6563b7be3a2ef592
